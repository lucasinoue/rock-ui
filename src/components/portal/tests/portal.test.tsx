import { render, screen } from '@testing-library/react';

import * as React from 'react';
import { Portal, PortalManager } from '../';

test('should render portal', async () => {
  const { asFragment } = render(
    <Portal>
      <>This is a portal 1</>
      <Portal>This is a portal 2</Portal>
    </Portal>
  );

  expect(asFragment()).toMatchSnapshot();
});

test('should render in a different node', () => {
  render(
    <PortalManager>
      <div data-testid="parent">
        <h1 data-testid="child-1">Foo</h1>
        <Portal>
          <h1 data-testid="child-2">Foo</h1>
        </Portal>
      </div>
    </PortalManager>
  );

  const parent = screen.getByTestId('parent');

  const child1 = screen.getByTestId('child-1');
  const child2 = screen.getByTestId('child-2');

  expect(parent).toContainElement(child1);
  expect(parent).not.toContainElement(child2);
});

test('should render into a custom container', () => {
  const Custom = () => {
    const ref = React.useRef<any>(null);
    return (
      <PortalManager>
        <div data-testid="container" ref={ref} />
        <Portal containerRef={ref}>
          <h1 data-testid="heading">Hello world</h1>
        </Portal>
      </PortalManager>
    );
  };

  const tools = render(<Custom />);

  const heading = tools.getByTestId('heading');
  const container = tools.getByTestId('container');

  expect(container).toContainElement(heading);
});
