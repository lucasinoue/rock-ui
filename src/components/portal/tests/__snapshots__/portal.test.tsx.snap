// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render portal 1`] = `<DocumentFragment />`;
